collection-loader.js

```diff
@@ -1,10 +1,18 @@
 const get = require("lodash/get");
 const flatMap = require("lodash/flatMap");
 
-function loadCollectionItems(client, collections, { storyFields, storyLimits, defaultNestedLimit }) {
+function loadCollectionItems(
+  client,
+  collections,
+  { storyFields, storyLimits, defaultNestedLimit, customLayouts = [] }
+) {
   const bulkRequestBody = collections.reduce((acc, collection, index) => {
-    let limit = storyLimits[get(collection, ["associated-metadata", "layout"])];
-
+    let limit;
+    if (customLayouts[index]) {
+      limit = storyLimits[customLayouts[index]];
+    } else {
+      limit = storyLimits[get(collection, ["associated-metadata", "layout"])];
+    }
```

Comment:
let limit = storyLimits[get(collection, ["associated-metadata", "layout"])];
if (customLayouts[index]) {
      limit = storyLimits[customLayouts[index]];
    }
    
    why not this ?